docs/source/contributing.rst

```diff
@@ -3,6 +3,112 @@
 Contributing
 ============
 
-Work in progress. In the mean time, most of the concepts in the
-`pvlib python Contributing Guide <https://pvlib-python.readthedocs.io/en/latest/contributing.html>`_
-apply here.
\ No newline at end of file
+We welcome and encourage community contributions to the Solar Forecast
+Arbiter. This guide aims to provide potential contributors with a better
+idea of how to prepare their code for the solarforecastarbiter-core and
+what to expect from the project maintainers.
+
+This guide focuses on code contributions, but we equally value
+contributions such as issue reports, code review, and documentation
+suggestions.
+
+Nomenclature
+------------
+
+Contributors should review the `glossary of terms <https://github.com/SolarArbiter/solarforecastarbiter-core/wiki/Glossary>`_
+and the guidance for
+`variable names <https://github.com/SolarArbiter/solarforecastarbiter-core/wiki/Variable-names>`_.
+
+Code style
+----------
+
+solarforecastarbiter-core follows the `PEP 8 -- Style Guide for Python Code
+<https://www.python.org/dev/peps/pep-0008/>`_. Maximum line length for code
+is 79 characters.
+
+Code must be compatible with Python 3.7 and above.
+
+Set your editor to strip extra whitespace from line endings. This
+prevents the git commit history from becoming cluttered with whitespace
+changes.
+
+The majority of the solarforecastarbiter-core code consists of
+functions that accept primatives or pandas.Series. Some higher-level
+functions accept :py:mod:`~solarforecastarbiter.datamodel` objects.
+Avoid writing new classes and methods.
+
+Exceptions
+----------
+
+Code should raise an exception if a problem occurs and a solution is not
+certain to be correct in all situations.
+
+Logging
+-------
+
+Logging is encouraged in higher-level functions.
+
+DataFrames and Series
+---------------------
+
+The solarforecastarbiter-core repository focuses on analyses of time
+series data. Therefore, it is natural for the code to expect and return
+pandas objects with DatetimeIndexes. In most cases, code should expect
+and return one or more pandas.Series rather than a single
+pandas.DataFrame. A function that requires multiple Series is
+self-documenting by virtue of the names of the parameters. Providing or
+expecting the wrong data from such a function raises an exception at the
+API level, where it belongs. A function that expects a single DataFrame
+must carefully document the required keys, providing the wrong data will
+result in a problem deep within the function, and expecting the wrong
+data will result in a problem far away from the function.
```

Comment:
editorial: thanks for explaining the preference for Series over Dataframes, really helpful to understand the reasoning.